crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -1,27 +1,39 @@
 //! Entity handling types.
 //!
-//! In Bevy ECS, there is no monolithic data structure for an entity. Instead, the [`Entity`]
-//! `struct` is just a *generational index* (a combination of an ID and a generation). Then,
-//! the `Entity` maps to the specific [`Component`s](crate::component::Component). This way,
-//! entities can have meaningful data attached to it. This is a fundamental design choice
-//! that has been taken to enhance performance and usability.
+//! In the ECS paradigm, the **entity** is an abstract and elementary concept. It is closely related
```

Comment:
This takes too long to explain what an entity is.

Reply:
> An entity exclusively owns zero or more component instances, all of different types, and can dynamically acquire or lose them over its lifetime.

This explanation rocks; try and cut the fluff before this.

Reply:
I'm having a hard time about understanding what you don't like about the first two sentences. About the first one, I like it because it gives off the (correct) idea that entities are not like objects in OOP.

In OOP there are those monolithic concrete types, where to each conceptual object corresponds an equivalent description in the code. In ECS, the object (called entity) is abstracted away, and its data is scattered among components, “held together” just by a handle. &mdash; Obviously we can't put the OOP vs ECS in the API docs, it's more of a Book thing, as it explains a design choice (arguably an entire philosophy).

I agree that the second sentence (“It is closely related to components”) can be removed, as the third one shows very well the relationship between the two.

Reply:
>  I like it because it gives off the (correct) idea that entities are not like objects in OOP.

This is an important idea, but I think it's out of place here. I find that framing things by what they're not is generally ineffective; that sort of clarification belongs later.

I also think that "abstract and elementary" doesn't add a lot of value, the language is very academic.

Reply:
OK, you convinced me. I only left the relationship between an entity and its components. The lack of a definition gives itself the suggestion that entity is indeed an elementary concept.